microsetta_admin/server.py

```diff
@@ -774,6 +775,220 @@ def scan():
                                       recorded_type)
 
 
+def _get_color_code():
+    # val = '0123456789ABCDEF'
+    # code = ["#"+''.join([random.choice(val) for i in range(6)])]
+    codes = ["#C0C0C0", "#808080", "#800000", "#FF0000", "#800080", "#FF00FF",
+             "#008000", "#00FF00", "#FFd700", "#000080", "#0000FF", "#00FFFF",
+             "#FFE4C4", "#7FFFD4", "#DEB887", "#FF7F50", "#6495ED", "#FF8C00",
+             "#FF1493", "#00BFFF", "#1E90FF", "#DAA520", "#4B0082", "#F0E68C",
+             "#90EE90", "#778899", "#FFA500", "#BC8F8F", "#D8BFD8", "#DCDCDC"]
+
+    return codes[random.randrange(0, len(codes)-1)]
+
+
+def _get_legends(criteria):
+
+    dict = {}
+    if criteria not in session:
+        status, fields = APIRequest.get(
+            '/api/admin/barcode_query_fields')
+
+        if status == 200:
+            for obj in fields:
+                if obj["label"] == criteria:
+                    for val in obj["values"]:
+                        color_code = None
+                        if criteria == "Project":
+                            if not ('-' in obj["values"][val]):
+                                color_code = _get_color_code()
+                        elif "Sample" in criteria:
+                            color_code = _get_color_code()
+
+                        if color_code is not None:
+                            while color_code in dict.values():
```

Comment:
It looks like this has the potential to go into an infinite loop if the supply of defined colors is exhausted. Can that be handled more gracefully?